internal/ir/infra.go

```diff
@@ -0,0 +1,65 @@
+package ir
+
+import (
+	cfgv1a1 "github.com/envoyproxy/gateway/api/config/v1alpha1"
+)
+
+// Infra defines managed infrastructure.
+type Infra struct {
+	// Proxy defines managed proxy infrastructure.
+	Proxy *ProxyInfra
+}
+
+// ProxyInfra defines managed proxy infrastructure.
+type ProxyInfra struct {
+	// TODO: Figure out how to represent metadata in the IR.
+	// xref: https://github.com/envoyproxy/gateway/issues/173
+	//
+	// Name is the name used for managed proxy infrastructure.
+	Name string
+	// Namespace is the namespace used for managed proxy infrastructure.
+	Namespace string
+	// OwnerRefs is a list of objects used to generate this object.
+	OwnerRefs []ObjectReference
+	// Config defines user-facing configuration of the managed proxy infrastructure.
+	Config *cfgv1a1.EnvoyProxy
```

Comment:
https://github.com/envoyproxy/gateway/blob/0ddb112929b427c1a079c320129cf64e3ce98379/api/config/v1alpha1/envoyproxy_types.go#L20 is currently empty, prefer if an issue was raised tracking proxy overlays applied by the user and how this would differ for K8s and non K8s cases

Reply:
Yes, `EnvoyProxySpec` is empty. I added it to the infra IR since we need to plumb user config to the infra manager.

Reply:
👍 prefer adding this field when we have the end to end workflow of applying such configs ready